frontstage/views/sign_in/sign_in.py

```diff
@@ -30,49 +36,40 @@ def login():
     if request.method == 'POST' and form.validate():
         username = request.form.get('username')
         password = request.form.get('password')
-        sign_in_data = {
-            "username": username,
-            "password": password
-        }
-        response = api_call('POST', app.config['SIGN_IN_URL'], json=sign_in_data)
 
-        # Handle OAuth2 authentication errors
-        if response.status_code == 401:
-            error_json = json.loads(response.text).get('error')
-            error_message = error_json.get('data', {}).get('detail')
-            if 'Unauthorized user credentials' in error_message:
+        try:
+            oauth2_token = oauth_controller.sign_in(username, password)
+        except OAuth2Error as exc:
+            error_message = exc.message
+            if BAD_AUTH_ERROR in error_message:
                 return render_template('sign-in/sign-in.html', form=form, data={"error": {"type": "failed"}})
-            elif 'User account locked' in error_message:
+            elif ACCOUNT_LOCKED_ERROR in error_message:
                 logger.debug('User account is locked on the OAuth2 server')
                 return render_template('sign-in/sign-in.trouble.html', form=form, data={"error": {"type": "account locked"}})
-            elif 'User account not verified' in error_message:
+            elif NOT_VERIFIED_ERROR in error_message:
                 logger.debug('User account is not verified on the OAuth2 server')
                 return render_template('sign-in/sign-in.account-not-verified.html', form=form,
                                        data={"error": {"type": "account not verified"}})
-            else:
-                logger.error('OAuth 2 server generated 401 which is not understood',
-                             oauth2error=error_message)
-                return render_template('sign-in/sign-in.html', form=form, data={"error": {"type": "failed"}})
-
-        if response.status_code != 200:
-            logger.error('Failed to sign in')
-            raise ApiError(response)
+            logger.error('OAuth 2 server generated 401 which is not understood',
+                         oauth2error=error_message)
+            return render_template('sign-in/sign-in.html', form=form, data={"error": {"type": "failed"}})
```

Comment:
An explicit else, while not required, would enhance the clarity of this if/else block.

Reply:
@JamesGardiner Not a fan of an extra `else` but if it reads clunky then I'll add it here.

Reply:
Nope, just a suggestion 👍